tests/kernel/semaphore/semaphore/src/main.c

```diff
@@ -1327,11 +1327,11 @@ void test_main(void)
 			 ztest_user_unit_test(test_sem_give_from_thread),
 			 ztest_user_unit_test(test_sem_take_no_wait),
 			 ztest_user_unit_test(test_sem_take_no_wait_fails),
-			 ztest_1cpu_user_unit_test(test_sem_take_timeout_fails),
-			 ztest_user_unit_test(test_sem_take_timeout),
-			 ztest_1cpu_user_unit_test(test_sem_take_timeout_forever),
+			 ztest_user_unit_test(test_sem_take_timeout_fails),
+			 ztest_1cpu_user_unit_test(test_sem_take_timeout),
```

Comment:
Why are you changing the `test_sem_take_timeout` to 1cpu?

Reply:
Hi, because based on the design of this test case "test_sem_take_timeout", 1cpu is necessary, I believe this not design for multi-core in the first place. It will fail in more than two core. The test logic here is to spawn a thread,  check its count,  then take a semaphore(the spawned thread is scheduled to run). But with multi-core, it would be scheduled right away before the check count, then the test case will fail.
So I make it 1cpu here. Of course, I can slightly change the test code to adapt to multi-core, but I suggest just keep it this way because the test goal is simple here. How do you think about it? Thanks for you review ^^

Reply:
Hi @dcpleung, do you think should I not make it 1cpu here? thanks~

Reply:
It has been working without 1cpu. So I think we should keep it that way, especially what you are trying to do in this PR is to enable testing on multiple CPUs.

Reply:
OK, got it, I think you are right. This makes it more consistent with this PR. I will update this, thanks for your suggestion!